### ISSUE TITLE
Fix S2225 FP: Support static virtual/abstract interface methods

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6307]
### Description

FP for S2225 (ToStringShouldNotReturnNull).

Using C#11 static virtual/abstract interface methods, the rule is erroneously raising an issue.

### Repro steps

```csharp
public interface SomeInterface
{
    static virtual string ToString()
    {
        return null; // Noncompliant FP
    }
}

public interface SomeOtherInterface
{
    static abstract string ToString();
}

public class SomeClass: SomeOtherInterface
{
    public static string ToString()
    {
        return null; // Noncompliant FP
    }
}

```

### Expected behavior

The rule should not raise an issue.

### Actual behavior

The rule does raise an issue.
